heatshrinkpy/core/encoder.py

```diff
@@ -254,7 +254,7 @@ def _find_longest_match(self, start: int, end: int, maxlen: int) -> Optional[Tup
                 continue
 
             for ml in range(1, maxlen + 1):
-                if self._buffer[pos + ml] != self._buffer[end + ml]:
+                if ml == maxlen or self._buffer[pos + ml] != self._buffer[end + ml]:
```

Comment:
I think there's no need to check for `ml == maxlen`. Just removing ` + 1` in the `range()` call is enough.